statsmodels/__init__.py

```diff
@@ -25,55 +26,29 @@
         # we have currently many ResourceWarnings in the datasets on python 3.4
         warnings.simplefilter("ignore", ResourceWarning)
 
-
-class NoseWrapper(Tester):
-    '''
-    This is simply a monkey patch for numpy.testing.Tester.
-
-    It allows extra_argv to be changed from its default None to ['--exe'] so
-    that the tests can be run the same across platforms.  It also takes kwargs
-    that are passed to numpy.errstate to suppress floating point warnings.
-    '''
-    def test(self, label='fast', verbose=1, extra_argv=['--exe'],
-             doctests=False, coverage=False, **kwargs):
-        ''' Run tests for module using nose
-
-        %(test_header)s
-        doctests : boolean
-            If True, run doctests in module, default False
-        coverage : boolean
-            If True, report coverage of NumPy code, default False
-            (Requires the coverage module:
-             http://nedbatchelder.com/code/modules/coverage.html)
-        kwargs
-            Passed to numpy.errstate.  See its documentation for details.
-        '''
-
-        # cap verbosity at 3 because nose becomes *very* verbose beyond that
-        verbose = min(verbose, 3)
-
-        from numpy.testing import utils
-        utils.verbose = verbose
-
-        if doctests:
-            print("Running unit tests and doctests for %s" % self.package_name)
-        else:
-            print("Running unit tests for %s" % self.package_name)
-
-        self._show_system_info()
-
-        # reset doctest state on every run
-        import doctest
-        doctest.master = None
-
-        argv, plugins = self.prepare_test_args(label, verbose, extra_argv,
-                                               doctests, coverage)
-        from numpy.testing.noseclasses import NumpyTestProgram
-        with errstate(**kwargs):
-            simplefilter('ignore', category=DeprecationWarning)
-            t = NumpyTestProgram(argv=argv, exit=False, plugins=plugins)
-        return t.result
-test = NoseWrapper().test
+class PytestTester(object):
+    def __init__(self):
+        f = sys._getframe(1)
+        package_path = f.f_locals.get('__file__', None)
+        if package_path is None:
+            raise ValueError('Unable to determine path')
+        self.package_path = os.path.dirname(package_path)
+        self.package_name = f.f_locals.get('__name__', None)
+
+    def __call__(self, extra_args=None):
+        try:
+            import pytest
+            if not LooseVersion(pytest.__version__) >= LooseVersion('3.0'):
+                raise ImportError
+            extra_args = [] if extra_args is None else extra_args
+            cmd = [self.package_path] + extra_args
+            print('Running pytest '+ ' '.join(cmd))
+            errno = pytest.main(cmd)
+            sys.exit(errno)
```

Comment:
I think it shouldn't exit, the same way as before. 
i.e. I was running different tests in the same python session